src/scalarstats.jl

```diff
@@ -532,7 +532,12 @@ Compute the entropy of a collection of probabilities `p`,
 optionally specifying a real number `b` such that the entropy is scaled by `1/log(b)`.
 Elements with probability 0 or 1 add 0 to the entropy.
 """
-entropy(p) = -sum(pᵢ -> iszero(pᵢ) ? zero(pᵢ) : pᵢ * log(pᵢ), p)
+function entropy(p)
+    if isempty(p)
+        throw(ArgumentError("empty collections of probabilities are not supported"))
```

Comment:
Maybe more explicit?
```suggestion
        throw(ArgumentError("empty collections are not supported as they do not " *
                            "represent a proper probability distribution""))
```